File: UniFarm/Capstone.UniFarm.Domain/Models/Category.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Capstone.UniFarm.Domain.Models
{
    [Table("Category")]
    public partial class Category
    {
        public Category()
        {
            Products = new HashSet<Product>();
        }

        [Key]
        public Guid Id { get; set; }
        [StringLength(255)]
        public string Name { get; set; } = null!;
        [StringLength(1000)]
        public string? Description { get; set; }
        [StringLength(255)]
        public string? Image { get; set; }
        [StringLength(100)]
        public string? Code { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? CreatedAt { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? UpdatedAt { get; set; }
        [StringLength(100)]
        public string? Status { get; set; }
        public int? DisplayIndex { get; set; }
        public double? SystemPrice { get; set; }
        public double? MinSystemPrice { get; set; }
        public double? MaxSystemPrice { get; set; }
        public double? Margin { get; set; }

        [InverseProperty(nameof(Product.Category))]
        public virtual ICollection<Product> Products { get; set; }
    }
}

File: FTAFE/src/core/models/category.ts
export type CreateCategory = Omit<Category, 'id' | 'createdAt' | 'updatedAt'>;
export type UpdateCategory = Omit<Category, 'id'>;
export type Category = {
    id: string;
    name: string;
    description: string;
    image: string;
    code: string;
    status: string;
    displayIndex: number;
    systemPrice: number;
    minSystemPrice: number;
    maxSystemPrice: number;
    margin: number;
    createdAt: string;
    updatedAt: string;
};
